tensorrt_llm/_torch/models/modeling_llama.py

```diff
@@ -686,7 +710,7 @@ def forward(
         hidden_states, hidden_states_to_save = self.norm(
             hidden_states, residual)
         assert isinstance(spec_metadata, Eagle3SpecMetadata)
-        spec_metadata.hidden_states.append(hidden_states_to_save)
+        spec_metadata.maybe_capture_hidden_states(1, hidden_states_to_save)
```

Comment:
Why is the layer_id set to 1 here? Overall I don't understand how you pass the intermediate target model states to the eagle draft model. Like where and how you pass them?

Reply:
You copy the hidden states into the spec metadata for a specified set of layer outputs. We collect the hidden states from the spec metadata later in the executor. 

As for the 1, it's because the eagle specification is weird. We already have a decoder layer with layer ID zero. But we don't want to capture the outputs from it when we run the draft model, we want to capture the outputs after an additional norm. So give it the label 1 instead of zero to avoid capturing the wrong states.

Reply:
Could you add these to the code as comments? It's hard to infer all this from the code.